### PR TITLE
new InvenSense icm20948 driver on SPI with AK09916 mag

### PR DESCRIPTION
**Pull request by dagar**

### CODE REVIEW

**Comment by mrpollo** [comment]
@dagar Is this PR blocked by other future changes?

**Comment by dakejahl** [comment]
image

This is good to merge.

**Comment by dagar** [comment]
Thanks @dakejahl

File: src/drivers/drv_sensor.h [hunk 1]
```diff
@@ -79,6 +79,7 @@
 #define DRV_IMU_DEVTYPE_ICM20649 0x25
 #define DRV_IMU_DEVTYPE_ICM42688P 0x26
 #define DRV_IMU_DEVTYPE_ICM40609D 0x27
+#define DRV_IMU_DEVTYPE_ICM20948 0x28
```
Comment (dagar) [inline src/drivers/drv_sensor.h:82]: Note, I wouldn't typically do this, but the IMU portion of the old icm20948 never worked properly in the first place.

**Pull request merged**